src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_SIZE = 16;
+    private static final int MULTIPLIER = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_SIZE];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = hash(key);
+        int index = hash & table.length - 1;
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        Node<K, V> newNode = new Node<>(key, value, hash, null);
+        newNode.next = table[index];
+        table[index] = newNode;
+        size++;
+        resize();
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> kvNode : table) {
+            Node<K, V> node = kvNode;
+            if (node == null) {
+                continue;
+            }
+            while (node != null) {
+                if (Objects.equals(key, node.key)) {
+                    return node.value;
+                }
+                node = node.next;
+            }
+        }
         return null;
```

Comment:
```suggestion
        int index = getIndex(key);
        Node<K, V> entry = table[index];
        while (entry != null) {
            if (Objects.equals(entry.key, key)) {
                return entry.value;
            }
            entry = entry.next;
        }
        return null;
```
and implement a method to get an index

Reply:
that's the problem. Because table is dynamic "int index = hash & table.length - 1;" won't work for "get" method. That's the reason I used approach of iterating through every bucket.

Reply:
You can implement getIndex in this way
```
private int getIndex(K key) {
        return hash(key) % table.length;
    }
```
and you approach that was proposed by mentor